mindsdb/libs/phases/type_deductor/type_deductor.py

```diff
@@ -244,13 +244,14 @@ def run(self, input_data):
         stats = defaultdict(dict)
         stats_v2 = defaultdict(dict)
 
-        # Really bad that these parameters are implicitly passed through lmd
-        # Perhaps sampling can be moved somewhere upwards,
-        # so that it can be reused by all downstream phases?
-        sample_df = sample_data(input_data.data_frame,
-                                self.transaction.lmd['sample_margin_of_error'],
-                                self.transaction.lmd['sample_confidence_level'],
-                                self.log)
+        self.transaction.sample_df = sample_data(
```

Comment:
You modified the class of the `input_data` object not the `transaction` object.

So `self.transaction.sample_df` needs to be `self.transaction.input_data.sample_df`  ( I believe double-check the naming and try running all tests locally first)